ethereum/eip712/encoding.go

```diff
@@ -262,53 +251,22 @@ func validatePayloadMessages(msgs []sdk.Msg) error {
 		return errors.New("unable to build EIP-712 payload: transaction does contain any messages")
 	}
 
-	var msgType string
 	var msgSigner sdk.AccAddress
 
 	for i, m := range msgs {
-		t, err := getMsgType(m)
-		if err != nil {
-			return err
-		}
-
 		if len(m.GetSigners()) != 1 {
```

Comment:
Q: Even though now we can have different msg types, shouldn't we check that the msgs types are valid/existent?

Reply:
This should be covered by Cosmos SDK's `checkTx`.